packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -0,0 +1,590 @@
+          : widget.displayStringForOption(nextSelection);
+      _textEditingController.value = TextEditingValue(
+        selection: TextSelection.collapsed(offset: selectionString.length),
+        text: selectionString,
+      );
+      widget.onSelected?.call(_selection);
+    });
+  }
+
+  // Hide or show the options overlay, if needed.
+  void _updateOverlay() {
+    if (_shouldShowOptions) {
+      final RenderBox renderBox = context.findRenderObject()! as RenderBox;
+      _floatingOptions?.remove();
+      _floatingOptions = OverlayEntry(
+        builder: (BuildContext context) {
+          return _FloatingOptions<T>(
+            optionsViewBuilder: widget.optionsViewBuilder,
+            fieldSize: renderBox.size,
+            layerLink: _optionsLayerLink,
+            onSelected: _select,
+            options: _options,
+          );
+        },
+      );
+      Overlay.of(context, rootOverlay: true)!.insert(_floatingOptions!);
+    } else if (_floatingOptions != null) {
+      _floatingOptions!.remove();
+      _floatingOptions = null;
+    }
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _textEditingController.addListener(_onChangedField);
+    SchedulerBinding.instance!.addPostFrameCallback((Duration _) {
+      _updateOverlay();
+    });
+  }
+
+  @override
+  void didUpdateWidget(AutocompleteCore<T> oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    SchedulerBinding.instance!.addPostFrameCallback((Duration _) {
+      _updateOverlay();
+    });
+  }
+
+  @override
+  void dispose() {
+    _textEditingController.removeListener(_onChangedField);
+    _floatingOptions?.remove();
+    _floatingOptions = null;
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      key: _fieldKey,
+      child: CompositedTransformTarget(
+        link: _optionsLayerLink,
+        child: widget.fieldViewBuilder(
+          context,
+          _textEditingController,
+          _onFieldSubmitted,
+        ),
+      ),
+    );
+  }
+}
+
+// The floating options, meant to be built inside of an Overlay. Will position
+// itself at the bottom of the field indicated by layerLink and fieldSize.
+class _FloatingOptions<T extends Object> extends StatelessWidget {
+  const _FloatingOptions({
+    Key? key,
+    required this.optionsViewBuilder,
+    required this.fieldSize,
+    required this.layerLink,
+    required this.onSelected,
+    required this.options,
+  }) : assert(optionsViewBuilder != null),
+       assert(fieldSize != null),
+       assert(layerLink != null),
+       assert(onSelected != null),
+       assert(options != null),
+       super(key: key);
+
+  final AutocompleteOptionsViewBuilder<T> optionsViewBuilder;
+  final Size fieldSize;
+  final LayerLink layerLink;
+  final AutocompleteOnSelected<T> onSelected;
+  final Iterable<T> options;
+
+  @override
+  Widget build(BuildContext context) {
+    return Positioned(
+      width: fieldSize.width,
```

Comment:
Do we document the fact that the options view will be same width as the field view? What If I want my options view to be wider than the field?

Reply:
Good point.  I've refactored this to be unopinionated about the width.  By default the options view can be as wide as the user wants.  If needed, they can make it more narrow.

Here is an example of wanting a narrow field and narrow options to match: https://github.com/justinmc/flutter-autocomplete-examples/blob/master/lib/core_basic_narrow_page.dart

Reply:
I like this, since as you pointed out with the example, one can force the input field and the options view to have the same width, but it's not required.